std/typecons.d

```diff
@@ -3966,6 +4006,62 @@ if (is(Interface == interface) && is(BaseClass == class))
     mixin(autoImplement_helper_.code);
 }
 
+///
+@system unittest
+{
+    interface PackageSupplier
+    {
+        int foo();
+        int bar();
+    }
+
+    static abstract class AbstractFallbackPackageSupplier : PackageSupplier
+    {
+        protected PackageSupplier default_, fallback;
+
+        this(PackageSupplier default_, PackageSupplier fallback)
+        {
+            this.default_ = default_;
+            this.fallback = fallback;
+        }
+
+        abstract int foo();
+        abstract int bar();
```

Comment:
For the record: this is a workaround against  https://issues.dlang.org/show_bug.cgi?id=2525